packaging/win/delugewin.spec

```diff
@@ -3,320 +3,128 @@
-          target_arch=None,
-          codesign_identity=None,
-          entitlements_file=None )
-
-h = Analysis([os.path.abspath(os.path.join(HOMEPATH,os.pardir,os.pardir)) + '\Scripts\deluge-web-script.py'],
-             pathex=[],
-             binaries=binaries,
-             datas=datas,
-             hiddenimports=hiddenimports,
-             hookspath=[],
-             hooksconfig={},
-             runtime_hooks=[],
-             excludes=[],
-             win_no_prefer_redirects=False,
-             win_private_assemblies=False,
-             cipher=block_cipher,
-             noarchive=False)
-pyzh = PYZ(h.pure, h.zipped_data,
-             cipher=block_cipher)
-exeh = EXE(pyzh,
-          h.scripts,
-          [],
-          exclude_binaries=True,
-          name='deluge-web',
-          debug=False,
-          bootloader_ignore_signals=False,
-          strip=False,
-          upx=True,
-          icon='../../deluge/ui/data/pixmaps/deluge.ico',
-          console=False,
-          disable_windowed_traceback=False,
-          target_arch=None,
-          codesign_identity=None,
-          entitlements_file=None )
-
-coll = COLLECT(exe,
-               a.binaries,
-               a.zipfiles,
-               a.datas,
-               exeb,
-               b.binaries,
-               b.zipfiles,
-               b.datas,
-               exec,
-               c.binaries,
-               c.zipfiles,
-               c.datas,
-               exed,
-               d.binaries,
-               d.zipfiles,
-               d.datas,
-               exee,
-               e.binaries,
-               e.zipfiles,
-               e.datas,
-               exef,
-               f.binaries,
-               f.zipfiles,
-               f.datas,
-               exeg,
-               g.binaries,
-               g.zipfiles,
-               g.datas,
-               exeh,
-               h.binaries,
-               h.zipfiles,
-               h.datas,
-               strip=False,
-               upx=True,
-               upx_exclude=[],
-               name='Deluge')
+# List of executables to produce
+executables = {
+  'deluge-script.pyw': {
+    'name': 'deluge',
+    'console': False,
+  },
+  'deluge-gtk-script.pyw': {
+    'name': 'deluge-gtk',
+    'console': False,
+  },
+  'deluge-console-script.py': {
+    'name': 'deluge-console',
+    'console': True,
+  },
+  'deluge-debug-script.py': {
+    'name': 'deluge-debug',
+    'console': True,
+  },
+
+  'deluged-script.py': {
+    'name': 'deluged',
+    'console': False,
+  },
+  'deluged-debug-script.py': {
+    'name': 'deluged-debug',
+    'console': True,
+  },
+
+  'deluge-web-debug-script.py': {
+    'name': 'deluge-web-debug',
+    'console': False,
+  },
+  'deluge-web-script.py': {
+    'name': 'deluge-web',
+    'console': True,
+  }
+}
+
+analysis = {}
+exe = {}
+coll = []
+
+# Perform analysis
+for e, d in executables.items():
+  runtime_hooks = []
+  if !d.console:
```

Comment:
I'm guessing this excludes exes with console? However we need to exclude from more than just that also the web exes as well. And does console here refer to the console window or the deluge console? Cause if its window console and not deluge console we need to exclude deluge console as well.

Reply:
Sorry if that was not clear what i mean is the deluge-web-script has console false so weould fall under this check but it is not a gtk exe and therefore doesn't need the hook, is it a big deal probabbly not but the way it is now we can control that so lets not go backwards? :)

Reply:
Please check with code:

```py
...
EXE(
  ...
  console=d['console'],
  ...)
...
```

You are right however, this can't be used this way.